dora-parser/src/parser.rs

```diff
@@ -523,16 +496,19 @@ impl<'a> Parser<'a> {
             pos,
             span,
             data_type: ty,
-            is_pub: modifiers.contains(Modifier::Pub),
+            annotation_usages,
         })
     }
 
-    fn parse_class(&mut self, modifiers: &Modifiers) -> Result<Class, ParseErrorAndPos> {
+    fn parse_class(
+        &mut self,
+        annotation_usages: AnnotationUsages,
+    ) -> Result<Class, ParseErrorAndPos> {
         let start = self.token.span.start();
-        let has_open = modifiers.contains(Modifier::Open);
-        let internal = modifiers.contains(Modifier::Internal);
-        let is_abstract = modifiers.contains(Modifier::Abstract);
-        let is_pub = modifiers.contains(Modifier::Pub);
+        let use_cannon = annotation_usages.iter().any(|annotation_usage| {
```

Comment:
Can you check whether we still need the `@cannon` annotation? I don't think we need it anymore.

Reply:
It's still used here:

```
    let code = match bc {
        CompilerName::Cannon => cannon::compile(vm, &fct, src, &type_params),
        CompilerName::Boots => boots::compile(vm, &fct, src, type_params),
    };
```

Reply:
Right, there are still quite some uses of `@cannon`. Let's keep them here.